x-pack/plugins/features/server/ui_capabilities_for_features.ts

```diff
@@ -65,6 +86,60 @@ function getCapabilitiesFromFeature(feature: Feature): FeatureCapabilities {
   return UIFeatureCapabilities;
 }
 
+function getCapabilitiesFromElasticsearchFeature(
+  feature: ElasticsearchFeature
+): FeatureCapabilities {
```

Comment:
Optional: Can't some part of the logic in `getCapabilitiesFromFeature` and `getCapabilitiesFromElasticsearchFeature` be extracted/factorized?

Reply:
> Optional: Can't some part of the logic in getCapabilitiesFromFeature and getCapabilitiesFromElasticsearchFeature be extracted/factorized?

I'll explore this. I kept them separate while building and testing this logic, but didn't take the time to try to consolidate after the fact.

Reply:
I was able to collapse these into a set of shared functions, which cuts down on the duplication significantly. Thanks for calling this out!